lib/index.js

```diff
@@ -99,25 +99,39 @@ Strategy.prototype.authenticate = function (req, options) {
 Strategy.prototype.authorizationParams = function(options) {
   var options = options || {};
 
-  var params = {};
-  if (options.connection && typeof options.connection === 'string') {
-    params.connection = options.connection;
-
-    if (options.connection_scope && typeof options.connection_scope === 'string') {
-      params.connection_scope = options.connection_scope;
+  var {connection, connection_scope, audience, prompt, login_hint, acr_values, nonce, max_age} = options;
+  delete options.connection;
+  delete options.connection_scope;
+  delete options.audience;
+  delete options.prompt;
+  delete options.login_hint;
+  delete options.acr_values;
+  delete options.nonce;
+  delete options.max_age;
```

Comment:
This will impact the calling object that is passed to the `authorizationParams ` method, I don't think this is expected as this might break existing functionality.

I am curious about the Node version we should support, any idea on this @davidpatrick ? As this might end up a lot cleaner depending on which version to support.

Reply:
I guess if we wanna keep that delete method, we could avoid closing it twice and replace optionsClose with params.

Reply:
True - let's see what version of Node.js must be supported.

Reply:
Currently the codebase supports as low as Node 6

Reply:
@frederikprijck @davidpatrick I updated the code as requested, and added a test to make sure the function stays pure (i.e. not touching the input options). So ready for review now 🙃

Reply:
We are still cloning the object twice, I am also not sure why we need the destructuring. 
I think we can support this with minimal changes:

```
var params = Object.assign({}, options);
delete params .connection;
// delete the other properties
// rest of the code stays untouched, e.g.:
if (options.connection && typeof options.connection === 'string') {
  params.connection = options.connection;
  if (options.connection_scope && typeof options.connection_scope === 'string') {
    params.connection_scope = options.connection_scope;
  }
}
...
```

Reply:
I updated with your suggestion, and you are right. Destructuring is not needed, is it ok now @frederikprijck?